server/events/working_dir.go

```diff
@@ -323,17 +322,32 @@ func (w *FileWorkspace) Delete(r models.Repo, p models.PullRequest) error {
 	return os.RemoveAll(w.repoPullDir(r, p))
 }
 
-// DeleteForWorkspace deletes the working dir for this workspace.
-func (w *FileWorkspace) DeleteForWorkspace(r models.Repo, p models.PullRequest, workspace string) error {
-	return os.RemoveAll(w.cloneDir(r, p, workspace))
+func (w *FileWorkspace) DeleteWorkspaceForPath(r models.Repo, p models.PullRequest, path string, workspace string) error {
+	// Find planfile for all projects in this workspace.
+	files, err := filepath.Glob(fmt.Sprintf("%s/%s/*::%s.tfplan", w.cloneDir(r, p), path, workspace))
+	if err != nil {
+		return err
+	}
+
+	// Remove all projects planfiles.
+	for _, f := range files {
+		if err := os.Remove(f); err != nil {
+			return err
+		}
+	}
+
+	// Remove TF_DATA_DIR for the workspace.
+	os.RemoveAll(fmt.Sprintf("%s/%s/.terraform-%s", w.cloneDir(r, p), path, workspace))
```

Comment:
Instead of hard coding this information, can we retrieve the exact `TF_DATA_DIR` to remove?

Reply:
It does seem that we do not need to set up the TF_DATA_DIR at all since Terraform will set it up if a workspace is defined automatically so in that case we will not have a TF_DATA_DIR exposed, no?

I do not think regular terraform will set this ENV variable when switching toa. workspace in the shell context

Reply:
If `TF_DATA_DIR` is not set, it defaults to `.terraform`, if it is set, it can be pulled from the env value.

I'm suggesting, instead of assuming the data dir on this line, we should have a way of retrieving it programmatically rather than guessing the path.

If the user sets the TF_DATA_DIR for instance to something that does not match the above line, then this line will fail.

Reply:
Ok, I did some testing and we really need the `TF_DATA_DIR` for parallel jobs.

But, it's not that easy to get the `TF_DATA_DIR` for this function, this is executed on the `atlantis unlock` command (and maybe in a few more places), so by the time we reach this function the `TF_DATA_DIR` env is long gone.

Also the functions that have access to the final result of the `TF_DATA_DIR` only return a string containing the output and are being called everywhere in the code, I can create a model to represent the result containing both the output and the `TF_DATA_DIR` (and even more stuff related to the command result), but it will take me a few more days to adjust the code and the tests.

Reply:
Uff I see what you are saying, and yes that will be a bunch more work.
Take all the time you need Fabiano.

Reply:
I changed to draft based on this thread. Yes please take all the time you need since this is a big change. Thank you for your contribution here